lib/mongodb.js

```diff
@@ -744,26 +749,42 @@ function buildNearFilter(query, near) {
     coordinates = near.near;
   }
 
-  query.where[near.key] = {
-    near: {
-      $geometry: {
-        coordinates: [coordinates.lng, coordinates.lat],
-        type: 'Point',
-      },
+  node.near = {
+    $geometry: {
+      coordinates: [coordinates.lng, coordinates.lat],
+      type: 'Point',
     },
   };
-};
+
+  if (near.maxDistance) {
+    node.near['$maxDistance'] = near.maxDistance;
+  }
+}
 
 function hasNearFilter(where) {
   if (!where) return false;
 
-  for (var k in where) {
-    if (where[k] && typeof where[k] === 'object' && where[k].near) {
-      return true;
+  var found = false;
+  var keyPath = [];
+  searchForNear(where);
+
+  function searchForNear(node) {
+    if (node && typeof node === 'object') {
+      var keys = Object.keys(node);
+      for (var i = 0; i < keys.length; i++) {
+        var ex = node[keys[i]];
+        keyPath.push(keys[i]);
+        if (ex && ex.near) {
```

Comment:
@superkhau I inherited this naming convention and wasn't sure how you guys feel about PRs changing the names of variables... I'll submit an update with something that makes more sense to me. If you have a suggestion ahead of that, please let me know. :)